### PR TITLE
improve adt docs

### PR DESCRIPTION
**Pull request by drewfeelsblue**

### CODE REVIEW

File: docs/src/main/tut/codecs/adt.md [hunk 1]
```diff
@@ -21,25 +21,27 @@ case class Qux(values: List[String]) extends Event
...
 object GenericDerivation {
   implicit val encodeEvent: Encoder[Event] = Encoder.instance {
-    case foo @ Foo(_) => foo.asJson
-    case bar @ Bar(_) => bar.asJson
-    case baz @ Baz(_) => baz.asJson
-    case qux @ Qux(_) => qux.asJson
+    case foo: Foo => foo.asJson
+    case bar: Bar => bar.asJson
+    case baz: Baz => baz.asJson
+    case qux: Qux => qux.asJson
```
Comment (zarthross) [inline docs/src/main/tut/codecs/adt.md:33]: I like this is an improvement.

File: docs/src/main/tut/codecs/adt.md [hunk 2]
```diff
@@ -21,25 +21,27 @@ case class Qux(values: List[String]) extends Event
...
   implicit val decodeEvent: Decoder[Event] =
-    List[Decoder[Event]](
+    NonEmptyList.of[Decoder[Event]](
       Decoder[Foo].widen,
       Decoder[Bar].widen,
       Decoder[Baz].widen,
       Decoder[Qux].widen
-    ).reduceLeft(_ or _)
+    ).reduceK
 }
```
Comment (zarthross) [inline docs/src/main/tut/codecs/adt.md]: I don't think for the target audience of our documentation, that the use of `reduceK` over `reduceLeft(_ or _)` actually makes it any more clear on how we might compose decoders.

**Review by zarthross — CHANGES_REQUESTED** [review]
We'd be happy to merge in the change around the pattern match if you can remove the other change. Thanks for the contribution.

File: docs/src/main/tut/codecs/adt.md (hunk 2 above)
Comment (drewfeelsblue) [reply docs/src/main/tut/codecs/adt.md]: no problems, I'll remove it. I just believed that the docs should provide with, I'd say, the most "correct" code (and here passing the function explicitly is kind of redundant)

Comment (drewfeelsblue) [reply docs/src/main/tut/codecs/adt.md]: moreover, this code (with the `reduceLeft` on the list) can be considered as erroneous by the scapegoat's `UnsafeTraversableMethods` inspection

Comment (drewfeelsblue) [reply docs/src/main/tut/codecs/adt.md]: removed

Comment (zarthross) [reply docs/src/main/tut/codecs/adt.md]: Thanks for updating it.  I didn't think about the linter scapegoat, and the `UnsafeTraversableMethods`.  I think switching to a `NonEmptyList` would fix that specific issue, and showing another example with `reduceK` could be fine.  I just want to make sure our average audience understands at a glance what is going on.

**Review by zarthross — APPROVED** [review]

**Review by zmccoy — APPROVED** [review]

**Pull request merged**